src/components/reactcomponents/AddNewIcon.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import AddNewModal from './AddNewModal.js';
+
+/**
+ * Component for Add button.
+ * Handles opening/closing of the dialog box.
+ **/
+
+export default class AddNewIcon extends React.Component{
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: false
+    };
+  }
+
+  openModalDialog() {
```

Comment:
Typically if you have a function that is going to only be used by the component itself, i.e. private functions, you should put an underscore before the function name. `openModalDialog` to `_openModalDialog`